configure.ac

```diff
@@ -3974,7 +3974,18 @@ AC_SEARCH_LIBS(nanosleep, rt)
 
 AC_SEARCH_LIBS(dlopen, dl)
 old_ldflags="${LDFLAGS}"
-LDFLAGS="${LDFLAGS} -Wl,-export-dynamic"
+# GNU specific option, this confuses IBM ld, but do offer alternatives when possible
+if test $lt_cv_prog_gnu_ld = yes; then
+	LDFLAGS="${LDFLAGS} -Wl,-export-dynamic"
```

Comment:
not sure if that works with the linker on macOS too.

Reply:
I wasn't sure if LLVM's linker supports it; has Apple moved to that yet? AFAIK, they're still using GNU binutils even with clang. In any case, this could be extended to check for LLVM's linker and use the proper flag (maybe the same as GNU's) or any other non-GNU linker, really.

Reply:
I believe Apple has a very old GNU binutils fork, forked in Next days. There is lld but I don't think they use it.

Reply:
Apple doesn’t currently use either `lld` or GNU binutils for its linker. Apple’s linker is its own Open Source [ld64](https://opensource.apple.com/source/ld64/).